server/src/main/java/com/scalar/db/server/DistributedStorageService.java

```diff
@@ -69,8 +70,16 @@ public void get(GetRequest request, StreamObserver<GetResponse> responseObserver
 
           Get get = ProtoUtils.toGet(request.getGet(), metadata);
           Optional<Result> result = storage.get(get);
+
           GetResponse.Builder builder = GetResponse.newBuilder();
-          result.ifPresent(r -> builder.setResult(ProtoUtils.toResult(r)));
+
+          // For backward compatibility
+          if (ProtoUtils.isRequestFromOldClient(request.getGet())) {
+            result.ifPresent(r -> builder.setResult(ProtoUtils.toResultWithValue(r)));
+          } else {
+            result.ifPresent(r -> builder.setResult(ProtoUtils.toResult(r)));
+          }
```

Comment:
If the request is from an old client, we need to return `Result` with the deprecated `Value` for backward compatibility.